src/Illuminate/Macroable/Traits/Macroable.php

```diff
@@ -31,18 +31,28 @@ public static function macro($name, $macro)
     /**
      * Mix another object into the class.
      *
-     * @param  object  $mixin
+     * @param  object|iterable  $mixin
      * @param  bool  $replace
      * @return void
      *
      * @throws \ReflectionException
      */
     public static function mixin($mixin, $replace = true)
     {
+        if (is_iterable($mixin)) {
+            foreach ($mixin as $macro) {
+                static::mixin($macro, $replace);
+            }
+
+            return;
+        }
+
         $methods = (new ReflectionClass($mixin))->getMethods(
             ReflectionMethod::IS_PUBLIC | ReflectionMethod::IS_PROTECTED
         );
 
+        $mixin = (new ReflectionClass($mixin))->newInstance();
```

Comment:
Should we only do this if $mixin is a string?

Reply:
Originally had an `is_string` check there but honestly just didn't see the point if just making a new instance would work in either use-case, it skips out on adding another if statement and didn't break the test suite. Can add it back if that's preferable.

Reply:
It does make a difference of for example they did this with their mixin: `::mixin((new MixinableClass)->enableSomeOption());`

Reply:
Wouldn't that be the same as passing in a closure directly to mixin? And that doesn't work now, throwing an `ArgumentCountError`, with the message of `Closure::bind() expects at least 2 arguments, 0 given`. 🤔

So I'm a bit confused as to why that was the example given.